project/data_analysis/python/split_nulls/get_split_nulls.py

```diff
@@ -0,0 +1,163 @@
+from pspipe_utils import consistency
+from pspy import so_dict, pspy_utils
+from itertools import combinations
+import matplotlib.pyplot as plt
+import sys
+import numpy as np
+import scipy.stats as ss
+import pickle as p
+
+d = so_dict.so_dict()
+d.read_from_file(sys.argv[1])
+
+ps_dir = "spectra"
+cov_dir = "split_covariances"
+
+output_dir = "plots/split_nulls"
+pspy_utils.create_directory(output_dir)
+
+surveys = d["surveys"]
+
+arrays = [f"{sv}_{ar}" for sv in surveys for ar in d[f"arrays_{sv}"]]
+n_splits = {ar: len(d[f"maps_{ar}"]) for ar in arrays}
+
+modes = ["TT", "TE", "ET", "EE"]
+
+multipole_range = {
+    "dr6_pa4_f150": {
+        "T": [1250, 8500],
+        "E": [500, 8500]
+    },
+    "dr6_pa4_f220": {
+        "T": [1000, 8500],
+        "E": [500, 8500]
+    },
+    "dr6_pa5_f090": {
+        "T": [800, 8500],
+        "E": [500, 8500]
+    },
+    "dr6_pa5_f150": {
+        "T": [800, 8500],
+        "E": [500, 8500]
+    },
+    "dr6_pa6_f090": {
+        "T": [600, 8500],
+        "E": [500, 8500]
+    },
+    "dr6_pa6_f150": {
+        "T": [600, 8500],
+        "E": [500, 8500]
+    }
+}
+
+chi2_dict = {}
+pte_dict = {}
+for ar in arrays:
+
+    chi2_dict[ar] = {m: [] for m in modes}
+    pte_dict[ar] = {m: [] for m in modes}
+    ns = n_splits[ar]
+    splits_id = [str(i) for i in range(ns)]
+
+    cross_splits = list(combinations(splits_id, 2))
+    split_diff_list = list(combinations(cross_splits, 2))
+    print(split_diff_list)
```

Comment:
I understand that for printing purpose you need to cast the `combinations` into `list` but for the rest of the code you do not need this cast. So I suggest `cross_splits` and `split_diff_list` remains `combinations` objects and only cast them for printing purpose. Not casting make the combinations faster.